### PR TITLE
feat(collator): add top shards to block debug info

### PR DESCRIPTION
**Pull request by serejkaaa512**

### CODE REVIEW

File: collator/src/collator/debug_info.rs [hunk 1]
```diff
@@ -13,6 +14,7 @@ pub struct BlockDebugInfo<'a> {
     pub state: &'a ShardStateUnsplit,
     pub processed_upto: &'a ProcessedUptoInfoStuff,
     pub mc_state_extra: Option<&'a McStateExtra>,
+    pub top_shard_blocks_info: Option<&'a FastHashMap<ShardIdent, Box<ShardDescription>>>,
```
Comment (SmaGMan) [inline collator/src/collator/debug_info.rs:17]: Let's rename to `mc_top_shards`

File: collator/src/collator/debug_info.rs [hunk 2]
```diff
@@ -248,3 +257,49 @@ impl std::fmt::Debug for DebugShardFeeCreated<'_> {
...
 }
+
+pub struct TopShardBlocksDebugInfo<'a>(pub &'a FastHashMap<ShardIdent, Box<ShardDescription>>);
+impl std::fmt::Debug for TopShardBlocksDebugInfo<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut s = f.debug_struct("TopShardBlocksDebugInfo");
+
+        for (shard_id, shard) in self.0.iter() {
+            s.field("shard_id", &DebugDisplay(shard_id)).field(
+                "shard_description",
```
Comment (SmaGMan) [inline collator/src/collator/debug_info.rs:268]: Let's rename to `shard_descr`

File: collator/src/collator/debug_info.rs [hunk 3]
```diff
@@ -248,3 +257,49 @@ impl std::fmt::Debug for DebugShardFeeCreated<'_> {
             .finish()
     }
 }
+
+pub struct TopShardBlocksDebugInfo<'a>(pub &'a FastHashMap<ShardIdent, Box<ShardDescription>>);
+impl std::fmt::Debug for TopShardBlocksDebugInfo<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut s = f.debug_struct("TopShardBlocksDebugInfo");
```
Comment (SmaGMan) [inline collator/src/collator/debug_info.rs:264]: Let's rename to `McTopShards`

File: collator/src/collator/debug_info.rs [hunk 4]
```diff
@@ -248,3 +257,49 @@ impl std::fmt::Debug for DebugShardFeeCreated<'_> {
...
+                "shard_description",
+                &TopShardBlocksDescriptionDebugInfo(shard),
+            );
+        }
+
+        s.finish()
+    }
+}
+
+pub struct TopShardBlocksDescriptionDebugInfo<'a>(pub &'a ShardDescription);
```
Comment (SmaGMan) [inline collator/src/collator/debug_info.rs:277]: Let's rename to `DebugShardDescription`

File: collator/src/collator/debug_info.rs [hunk 5]
```diff
@@ -248,3 +257,49 @@ impl std::fmt::Debug for DebugShardFeeCreated<'_> {
...
+
+impl std::fmt::Debug for TopShardBlocksDescriptionDebugInfo<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TopShardBlocksDescription")
```
Comment (SmaGMan) [inline collator/src/collator/debug_info.rs:281]: Let's rename to `DebugShardDescription`

File: collator/src/collator/debug_info.rs [hunk 6]
```diff
@@ -248,3 +257,49 @@ impl std::fmt::Debug for DebugShardFeeCreated<'_> {
...
+            .field("seqno", &DebugDisplay(self.0.seqno))
```
Comment (SmaGMan) [inline collator/src/collator/debug_info.rs:282]: Do not need to wrap simple types

File: collator/src/collator/debug_info.rs [hunk 7]
```diff
@@ -248,3 +254,41 @@ impl std::fmt::Debug for DebugShardFeeCreated<'_> {
...
+
+        s.finish()
+    }
+}
+
+pub struct DebugShardDescription<'a>(pub &'a ShardDescription);
+
+impl std::fmt::Debug for DebugShardDescription<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("DebugShardDescription")
```
Comment (SmaGMan) [inline collator/src/collator/debug_info.rs:276]: Rename here to `ShardDescription`

**Review by 0xdeafbeef — CHANGES_REQUESTED** [review]

File: collator/src/collator/build_block.rs [hunk 8]
```diff
@@ -413,6 +413,7 @@ impl CollatorStdImpl {
                 prev_ref: &prev_ref,
                 state: &new_observable_state,
                 processed_upto: &collation_data.processed_upto,
+                mc_top_shards: collation_data.shards().ok(),
```
[the remark this answers is not in the document]
Comment (0xdeafbeef) [reply collator/src/collator/build_block.rs:416]: But it's useless to return `Result<T>` if you always convert it to `Option<T>`. You allocate `Box<dyn Error>` without a need.

[the remark this answers is not in the document]
Comment (SmaGMan) [reply collator/src/collator/build_block.rs]: Now we use method that returns `Option<T>`.

**Review by SmaGMan — APPROVED** [review]

**Review by 0xdeafbeef — APPROVED** [review]

**Pull request merged**